### ISSUE TITLE
cross validation

### ISSUE THREAD

**Issue by ld-archer** [Leeds-MRG/Minos#192]
About time we did this. The plan for cross-validation is to do a model run where we split the final population into 2 groups. With one group, we can estimate transition models, then we can simulate the other group using these transition models and see if we can replicate the trajectories. For the start year, we could use the first year where all the information we need is available (some variables don't start until partway through the survey - e.g. nutrition quality, tobacco use, neighbourhood safety).

This issue is linked to 2 more:
#157 & #166 

#166 has to come first, where we need to adjust the current logic for nowcasting to use generate replenishing populations in the same way we do for a projection run, and to allow for subsetting the final stock population and e.g. passing a subset to transitions.

**Comment by ld-archer** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Cross Validation](https://trello.com/c/u60sbA1B/184-cross-validation)

**Comment by ld-archer** [comment]
Found another problem whilst looking into the job_sector stuff. I previously added -10 to missing values to be removed in the complete case script, which meant lots of people who were unemployed were being removed from the dataset. I have removed -10 from the missing values list, and changed these -10 values to 0 to avoid any further confusion. So now unemployed and some other small groups are assigned a value of 0 for job_sec, which is kept in the data throughout.

I dealt with the job_sector issue in a similar way. Previously, people with job_sec == 5 (small employers and own account) very rarely had a value for job_sector. To solve this, I have assigned those who had a job_sector value of -8 (inapplicable) as 0. The majority of people in this -8 group were unemployed or job_sec == 5, so we want to keep these people even if we run complete case.

With this fix, cross-validation baseline simulation is running without error. I have also created an R script in validation/ named cross_validation.Rmd. Currently the only things in there are comparisons of the mean of hh_income and SF_12 between cross-validation and raw data (one half of final_US). Some t-tests and box plots will follow.
